src/backend/InvenTree/plugin/api.py

```diff
@@ -352,21 +355,21 @@ def get_object(self):
         The URL provides the 'slug' of the plugin, and the 'key' of the setting.
         Both the 'slug' and 'key' must be valid, else a 404 error is raised
         """
-        key = self.kwargs['key']
+        setting_key = self.kwargs['setting']
 
         # Look up plugin
-        plugin = check_plugin(
-            plugin_slug=self.kwargs.get('plugin'), plugin_pk=self.kwargs.get('pk')
-        )
+        plugin = check_plugin(self.kwargs.pop('key', None), None)
```

Comment:
Question: Do we need to check for existence of `key` and raise a ValidationError?

Reply:
The `check_plugin` method already does this:

![image](https://github.com/inventree/InvenTree/assets/10080325/cc8771a6-7d07-421f-b93b-b29debea1145)

Reply:
ok I somehow got it to not trigger with my tests, will investigate that further; Only 2 tests two fix left: https://github.com/matmair/InvenTree/actions/runs/9086781305/job/24973262465